godeltaprof/internal/pprof/delta_heap.go

```diff
@@ -6,15 +6,40 @@ import (
 	"strings"
 )
 
+type heapPrevValue struct {
+	allocObjects int64
+}
+
+type heapAccValue struct {
+	allocObjects int64
+	inuseObjects int64
+}
+
 type DeltaHeapProfiler struct {
-	m profMap
+	m profMap[heapPrevValue, heapAccValue]
+	//todo consider adding an option to remove block size label and merge allocations of different size
 }
 
 // WriteHeapProto writes the current heap profile in protobuf format to w.
 func (d *DeltaHeapProfiler) WriteHeapProto(b ProfileBuilder, p []runtime.MemProfileRecord, rate int64) error {
 	values := []int64{0, 0, 0, 0}
 	var locs []uint64
-	for _, r := range p {
+	for i := range p {
+		r := &p[i]
+		if r.AllocBytes == 0 && r.AllocObjects == 0 && r.FreeObjects == 0 && r.FreeBytes == 0 {
+			// it is a fresh bucket and it will be published after next 1-2 gc cycles
+			continue
+		}
+		var blockSize int64
+		if r.AllocObjects > 0 {
+			blockSize = r.AllocBytes / r.AllocObjects
+		}
+		entry := d.m.Lookup(stack(r.Stack0[:]), uintptr(blockSize))
```

Comment:
Why we're not using the standard `r.Stack()` here?

Reply:
<img width="629" alt="Screenshot 2024-06-03 at 6 53 40 PM" src="https://github.com/grafana/pyroscope-go/assets/331773/723eaf51-cb82-4a3c-81e0-defa89925823">

I think this was an worthless attempt to optimize this function - to avoid jumps into `duffcopy`. I is not needed. I am going to revert it to `r.Stack()`. Thanks.